py/dml/expr.py

```diff
@@ -80,8 +83,28 @@ def coverity_markers(markers, site=None):
         site_linemark_nocoverity(site)
 
 
+# Allow linemarks to be generated if dml.globals.linemarks_enabled is True
+# and the current output is FileOutput
+# This context manager generates a line directive reset when left unless
+# linemarks were already allowed, or manual_reset is True
+@contextmanager
+def allow_linemarks(manual_reset=False):
+    prev_linemarks = dml.globals.linemarks
+    curr_output = output.current()
+    is_file_output = isinstance(curr_output, FileOutput)
+    assert is_file_output >= prev_linemarks
+    assert dml.globals.linemarks_enabled >= prev_linemarks
```

Comment:
can you plese express this in first order logic instead?

Reply:
implications are first-order logic! It's not my fault that "X implies Y" is spelled `X <= Y`!

Reply:
I find it more readable to use conventional boolean operators, preferably on DNF or CNF.  like `is_file_output or not prev_linemarks`. And in this case, I think you can collapse both lines to `dml.globals.linemarks or not prev_linemarks`.

Reply:
I've fixed this in the latest commit, forgot to resolve it.